File: Plugins/Orimath.Basics.View/Controls/LineControl.cs
using System.ComponentModel;
using System.Windows;
using System.Windows.Controls;
using System.Windows.Shapes;

namespace Orimath.Basics.View.Controls
{
    public class LineControl : Control
    {
        [TypeConverter(typeof(DoubleConverter))]
        public double X1 { get => (double)GetValue(X1Property); set => SetValue(X1Property, value); }
        public static readonly DependencyProperty X1Property = Line.X1Property.AddOwner(typeof(LineControl));

        [TypeConverter(typeof(DoubleConverter))]
        public double X2 { get => (double)GetValue(X2Property); set => SetValue(X2Property, value); }
        public static readonly DependencyProperty X2Property = Line.X2Property.AddOwner(typeof(LineControl));

        [TypeConverter(typeof(DoubleConverter))]
        public double Y1 { get => (double)GetValue(Y1Property); set => SetValue(Y1Property, value); }
        public static readonly DependencyProperty Y1Property = Line.Y1Property.AddOwner(typeof(LineControl));

        [TypeConverter(typeof(DoubleConverter))]
        public double Y2 { get => (double)GetValue(Y2Property); set => SetValue(Y2Property, value); }
        public static readonly DependencyProperty Y2Property = Line.Y2Property.AddOwner(typeof(LineControl));

        [TypeConverter(typeof(DoubleConverter))]
        public double StrokeThickness { get => (double)GetValue(StrokeThicknessProperty); set => SetValue(StrokeThicknessProperty, value); }
        public static readonly DependencyProperty StrokeThicknessProperty = Line.StrokeThicknessProperty.AddOwner(typeof(LineControl));

        static LineControl()
        {
            DefaultStyleKeyProperty.OverrideMetadata(
                typeof(LineControl),
                new FrameworkPropertyMetadata(typeof(LineControl)));
        }
    }
}

File: Plugins/Orimath.Basics.View/Controls/LineControl.fs
namespace Orimath.Basics.View.Controls
open System.ComponentModel
open System.Windows
open System.Windows.Controls
open System.Windows.Shapes

type LineControl() =
    inherit Control()
    static let mutable init = false
    do
        if not init then
            init <- true
            FrameworkElement.DefaultStyleKeyProperty.OverrideMetadata(typeof<LineControl>, new FrameworkPropertyMetadata(typeof<LineControl>))

    [<TypeConverter(typeof<DoubleConverter>)>]
    member this.X1 with get() = this.GetValue(LineControl.X1Property) :?> float and set (v: float) = this.SetValue(LineControl.X1Property, box v)
    static member val X1Property = Line.X1Property.AddOwner(typeof<LineControl>)

    [<TypeConverter(typeof<DoubleConverter>)>]
    member this.X2 with get() = this.GetValue(LineControl.X2Property) :?> float and set (v: float) = this.SetValue(LineControl.X2Property, box v)
    static member val X2Property = Line.X2Property.AddOwner(typeof<LineControl>)

    [<TypeConverter(typeof<DoubleConverter>)>]
    member this.Y1 with get() = this.GetValue(LineControl.Y1Property) :?> float and set (v: float) = this.SetValue(LineControl.Y1Property, box v)
    static member val Y1Property = Line.Y1Property.AddOwner(typeof<LineControl>)

    [<TypeConverter(typeof<DoubleConverter>)>]
    member this.Y2 with get() = this.GetValue(LineControl.Y2Property) :?> float and set (v: float) = this.SetValue(LineControl.Y2Property, box v)
    static member val Y2Property = Line.Y2Property.AddOwner(typeof<LineControl>)

    [<TypeConverter(typeof<DoubleConverter>)>]
    member this.StrokeThickness with get() = this.GetValue(LineControl.StrokeThicknessProperty) :?> float and set (v: float) = this.SetValue(LineControl.StrokeThicknessProperty, box v)
    static member val StrokeThicknessProperty = Line.StrokeThicknessProperty.AddOwner(typeof<LineControl>)
